crates/exex/src/serde.rs

```diff
@@ -0,0 +1,226 @@
+            .iter_identifiers()
+            .filter(|(key, value)| {
+                key.contains(struct_name) &&
+                    key.split('.').last() == struct_name.split('.').last() &&
+                    value.type_ == expected_type
+            })
+            .map(|(_, value)| value)
+            .collect::<Vec<_>>();
+
+        // Match on the number of found identifiers
+        match identifiers.len() {
+            // No identifiers found
+            0 => Err(KakarotSerdeError::IdentifierNotFound {
+                struct_name: struct_name.to_string(),
+                expected_type,
+            }),
+            // Exactly one identifier found, return it
+            1 => Ok(identifiers[0].clone()),
+            // More than one identifier found
+            count => Err(KakarotSerdeError::MultipleIdentifiersFound {
+                struct_name: struct_name.to_string(),
+                expected_type,
+                count,
+            }),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use cairo_vm::types::{layout_name::LayoutName, program::Program};
+
+    fn setup_kakarot_serde() -> KakarotSerde {
+        // Load the valid program content from a JSON file
+        let program_content = include_bytes!("../../../cairo/programs/os.json");
+
+        // Create a Program instance from the loaded bytes, specifying "main" as the entry point
+        let program = Program::from_bytes(program_content, Some("main")).unwrap();
+
+        // Initialize a CairoRunner with the created program and default parameters
+        let runner = CairoRunner::new(&program, LayoutName::plain, false, false).unwrap();
+
+        // Return an instance of KakarotSerde
+        KakarotSerde { runner, memory: Memory::new() }
+    }
+
+    #[test]
+    fn test_program_identifier_valid() {
+        // Setup the KakarotSerde instance
+        let kakarot_serde = setup_kakarot_serde();
+
+        // Check if the identifier "main" with expected type "function" is correctly retrieved
+        assert_eq!(
+            kakarot_serde.get_identifier("main", Some("function".to_string())).unwrap(),
+            Identifier {
+                pc: Some(3478),
+                type_: Some("function".to_string()),
+                value: None,
+                full_name: None,
+                members: None,
+                cairo_type: None
+            }
+        );
+
+        // Check if the identifier "__temp0" with expected type "reference" is correctly retrieved
+        assert_eq!(
+            kakarot_serde.get_identifier("__temp0", Some("reference".to_string())).unwrap(),
+            Identifier {
+                pc: None,
+                type_: Some("reference".to_string()),
+                value: None,
+                full_name: Some("starkware.cairo.common.memcpy.memcpy.__temp0".to_string()),
+                members: None,
+                cairo_type: Some("felt".to_string())
+            }
+        );
```

Comment:
I would do this test because it doesn't mean anything for a cairo dev, it's auto generated naming

Reply:
This is just to assert a complex case and to be sure all the fields are well captured, do you want me to remove?